### PR TITLE
Send WSL2 version with Upgrade Responder query when on Windows

### LINKED ISSUE #4561: Add `wslVersion` field to information Rancher Desktop sends to Upgrade Responder
In the future, we may want to make some versions of Rancher Desktop supported/unsupported depending on which version of WSL2 a user has installed. However, it is not easy to get the WSL version in all cases. So, we want to get the WSL version if possible and include it in the information that is sent with WSL. If it is not feasible to get the WSL version, we want to omit it.

Epic: #3925

### PR DESCRIPTION
**Pull request by adamkpickering**
Closes #4561.

### CODE REVIEW

File: pkg/rancher-desktop/main/update/LonghornProvider.ts [hunk 1]
```diff
@@ -245,6 +247,47 @@ async function getPlatformVersion(): Promise<string> {
...
+// installed simultaneously. If this is the case, the application version
+// always takes precedence.
+//
+// It is difficult and not useful to get the version of any installed
+// in-box version of WSL. So, this function tries to get the version
+// of an installed application version of WSL, and if it cannot find
+// it for any reason (i.e. WSL is not installed, only an in-box version
+// is installed), it returns undefined.
+export async function getWslVersion(): Promise<string | undefined> {
+  const wslPath = 'C:\Windows\system32\wsl.exe';
```
Comment (jandubois) [inline pkg/rancher-desktop/main/update/LonghornProvider.ts:268]: Don't hard-code `C:\Windows` use the `%SystemRoot%` environment variable:

```suggestion
  const wslPath = path.join(process.env.SystemRoot, 'system32', 'wsl.exe');
```

File: pkg/rancher-desktop/main/update/__tests__/LonghornProvider.spec.ts [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
   it('should return the latest version', async() => {
+    (spawnFile as jest.Mock).mockResolvedValue({
+      stdout: `WSL version: 1.2.5.0
+Kernel version: 5.15.90.1
+WSLg version: 1.0.51
+MSRDC version: 1.2.3770
+Direct3D version: 1.608.2-61064218
+DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
+Windows version: 10.0.19044.2846
+`,
```
Comment (jandubois) [inline pkg/rancher-desktop/main/update/__tests__/LonghornProvider.spec.ts:40]: Should this be defined as a constant on top? It looks like the same response is used 7 times in this file.

**Review by jandubois — CHANGES_REQUESTED** [review]
Looks fine, except for the hard-coding of `C:\Windows`.

**Comment by adamkpickering** [comment]
@jandubois I've done some testing and feel more confident about this PR now. I'm good with merging it if you are.

File: pkg/rancher-desktop/main/update/LonghornProvider.ts [hunk 3]
```diff
@@ -245,6 +247,49 @@ async function getPlatformVersion(): Promise<string> {
...
+// installed simultaneously. If this is the case, the application version
+// always takes precedence.
+//
+// It is difficult and not useful to get the version of any installed
+// in-box version of WSL. So, this function tries to get the version
+// of an installed application version of WSL, and if it cannot find
+// it for any reason (i.e. WSL is not installed, only an in-box version
+// is installed), it returns undefined.
+export async function getWslVersion(): Promise<string | undefined> {
+  const args = ['--version'];
```
Comment (ericpromislow) [inline pkg/rancher-desktop/main/update/LonghornProvider.ts:268]: The command `wsl --version` doesn't work on Windows 10. However, `windows --status` does work, and I think the second line gives the output we need:

```
Default Version: 2
```

The first line is `Default Distribution: Ubuntu` might also be helpful. Don't know.

On Windows 11, `wsl --version` also returns an error message and the usual help text. I haven't installed any WSLs on the Windows 11 VM, so the first line reads `Default Version: 2`

**Review by ericpromislow — CHANGES_REQUESTED** [review]
I don't know how your `wsl --version` gives processable output and mine, on both windows 10 and 11, don't, but this needs to be addressed.

Apart from a typo, the rest is fine (although I didn't test it live).

File: pkg/rancher-desktop/main/update/LonghornProvider.ts (hunk 3 above)
Comment (adamkpickering) [reply pkg/rancher-desktop/main/update/LonghornProvider.ts:268]: It sounds like you have the in-box version of WSL installed. Unless I missed something, `getWslVersion` accounts for this case - it should return `undefined`. I've updated the comment on `getWslVersion` to explain that in-box versions of WSL do not have a `--version` flag.

Comment (ericpromislow) [reply pkg/rancher-desktop/main/update/LonghornProvider.ts:269]: This explanation should go in a comment in the code, with a note on how the system behaves if it has the in-box version installed, and returns `undefined`. I also don't see the problem with running `wsl --status` to get the WSL version, unless I'm misunderstanding something, and `version` means something else.

Sorry, I'm still confused about this code.

**Review by ericpromislow — APPROVED** [review]
Adam explained the in-box vs app forms of WSL and how the upgrade server does (not) use the version field, so I'm fine with the PR

**Comment by adamkpickering** [comment]
I see @jandubois has changes requested, which is preventing this from being merged.

**Pull request merged**